src/components/searchBar/index.js

```diff
@@ -5,17 +5,11 @@ import { translate } from 'react-i18next';
 import { searchSuggestions } from './../../actions/search';
 import actionTypes from './../../constants/actions';
 import AutoSuggest from './../autoSuggest';
-import routes from './../../constants/routes';
 import styles from './searchBar.css';
 
 class Search extends React.Component {
-  shouldShowSearchBarOnMobile() {
-    const { pathname } = this.props.location;
-    return pathname.includes('explorer') && !pathname.includes(`${routes.explorer.path}${routes.search.path}`);
-  }
-
   render() {
-    return (<div className={`${styles.searchBar} search-bar-input ${this.shouldShowSearchBarOnMobile() ? styles.show : null}`}>
```

Comment:
Removing `shouldShowSearchBarOnMobile` makes problems in most pages that don't account for the bar being shown

E.g.:
![screen shot 2018-06-18 at 11 03 15](https://user-images.githubusercontent.com/1254342/41527393-51f2360a-72e7-11e8-9183-ae89b018cb90.png)
Here it even hides the "Close" button:
![screen shot 2018-06-18 at 11 02 49](https://user-images.githubusercontent.com/1254342/41527391-51c8ef02-72e7-11e8-957a-b5e22fbb514f.png)

Let's not worry about S breakpoint. In M breakpoint we can fit the Search input into the top bar.